topgun/k8s/baggageclaim_drivers_test.go

```diff
@@ -94,4 +96,43 @@ var _ = Describe("Baggageclaim Drivers", func() {
 			ShouldWork: true,
 		}),
 	)
+
+	Describe("with a real btrfs partition", func() {
+		It("successfully recreates the worker", func() {
+			By("deploying concourse with ONLY one worker and having the worker pod use the gcloud disk and format it with btrfs")
+
+			setReleaseNameAndNamespace("real-btrfs-disk")
+
+			helmDeployTestFlags := []string{
+				"--set=concourse.worker.baggageclaim.driver=btrfs",
+				"--set=worker.nodeSelector.nodeImage=ubuntu",
+				"--set=worker.replicas=1",
+				"--set=persistence.enabled=false",
+				"--set=worker.additionalVolumes[0].name=concourse-work-dir",
+				"--set=worker.additionalVolumes[0].gcePersistentDisk.pdName=disk-topgun-k8s-btrfs-test",
```

Comment:
ooh nice! I didn't know this could be done within the context of a statefulset - very handy!

Reply:
Yeah nothing in the docs say you can or can't, so we just tried it. Not sure what happens if there's more than 1 replica, guess is that one worker would be able to write to it and the others would have read-only access.